processor/deltatocumulativeprocessor/internal/data/add.go

```diff
@@ -65,25 +65,95 @@ func (dp Histogram) Add(in Histogram) Histogram {
 	return dp
 }
 
+type bounds struct {
+	lower int32
+	upper int32
+}
+
+// with is an accessory for Merge() to calculate ideal combined scale.
+func (b bounds) with(o bounds) bounds {
+	if o.empty() {
+		return b
+	}
+	if b.empty() {
+		return o
+	}
+	return bounds{
+		lower: min(b.lower, o.lower),
+		upper: max(b.upper, o.upper),
+	}
+}
+
+// empty indicates whether there are any values in a bounds.
+func (b bounds) empty() bool {
+	return b == bounds{}
+}
+
+// boundsAtScale is an accessory for Add() to calculate ideal combined scale.
+func (dp ExpHistogram) boundsAtScale(b expo.Buckets, scale int32) bounds {
+	if b.BucketCounts().Len() == 0 {
+		return bounds{}
+	}
+	shift := dp.Scale() - scale
+	a := expo.Abs(b)
+	return bounds{
+		lower: int32(a.Lower()) >> shift,
+		upper: int32(a.Upper()) >> shift,
+	}
+}
+
+// downscaleNeeded computes how much downscaling is needed by shifting the
+// upper and lower bounds until they are separated by no more than size.
+func downscaleNeeded(b bounds, size int) int32 {
+	var change int32
+
+	for b.upper-b.lower > int32(size) {
+		b.upper >>= 1
+		b.lower >>= 1
```

Comment:
```suggestion
		b.upper /= 2
		b.lower /= 2
```

it's the same and reads easier if you're not super familier with bitshifts